test/functional/p2p_sendrecon.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test SENDRECON message
+"""
+
+from test_framework.messages import (
+    msg_sendrecon,
+    msg_verack,
+    msg_version,
+    msg_wtxidrelay,
+)
+from test_framework.p2p import (
+    P2PInterface,
+    P2P_SERVICES,
+    P2P_SUBVERSION,
+    P2P_VERSION,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal
+
+class PeerNoVerack(P2PInterface):
+    def __init__(self, wtxidrelay=True):
+        super().__init__(wtxidrelay=wtxidrelay)
+
+    def on_version(self, message):
+        # Avoid sending verack in response to version.
+        # When calling add_p2p_connection, wait_for_verack=False must be set (see
+        # comment in add_p2p_connection).
+        if message.nVersion >= 70016 and self.wtxidrelay:
+            self.send_message(msg_wtxidrelay())
+
+class SendReconReceiver(P2PInterface):
+    def __init__(self):
+        super().__init__()
+        self.sendrecon_msg_received = None
+
+    def on_sendrecon(self, message):
+        self.sendrecon_msg_received = message
+
+def create_sendrecon_msg(initiator=True):
+    sendrecon_msg = msg_sendrecon()
+    sendrecon_msg.initiator = initiator
+    sendrecon_msg.responder = not initiator
+    sendrecon_msg.version = 1
+    sendrecon_msg.salt = 2
+    return sendrecon_msg
+
+class SendReconTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 1
+        self.extra_args = [['-txrecon']]
+
+    def run_test(self):
+        self.log.info('SENDRECON sent to an inbound')
+        peer = self.nodes[0].add_p2p_connection(SendReconReceiver(), send_version=True, wait_for_verack=True)
+        assert peer.sendrecon_msg_received
+        assert not peer.sendrecon_msg_received.initiator
+        assert peer.sendrecon_msg_received.responder
+        assert_equal(peer.sendrecon_msg_received.version, 1)
```

Comment:
6ec8628181bb9342c338d180acc96f5a48185527:

I think it's good practice to disconnect these peers after you're done with them, to minimize the risk of subtests affecting each other. (And for the others below)

```suggestion
        assert_equal(peer.sendrecon_msg_received.version, 1)
        peer.disconnect_peer()
```